features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/CPURequirementChecker.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.listFilesSafe
+import java.io.File
+
+internal class CPURequirementChecker(
+    private val minCPUCores: Int,
+    private val cpuDirFile: File = File(DIR_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Int? = null
+
+    override fun checkMinimumRequirement(): Boolean {
+        return readCPUCoreNumber() >= minCPUCores
```

Comment:
we can optimize a bit this (and for the memory checker as well) - if `minCpuCores` is 0 (same for the RAM size), it makes no sense to do the check and we simply can return `true`. Saves some time by avoiding redundant syscalls.